db-service/lib/cqn2sql.js

```diff
@@ -505,7 +505,13 @@ class CQN2SQLRenderer {
 
   static Functions = require('./cql-functions')
   func({ func, args }) {
-    args = (args || []).map(e => (e === '*' ? e : { __proto__: e, toString: (x = e) => this.expr(x) }))
+    args = (args || []).map(e => {
+      if (e === '*') return e
+      const arg = { __proto__: e, toString: (x = e) => this.expr(x) }
+      if ('xpr' in arg)
+        arg.xpr = arg.xpr.map(e => (typeof e === 'string' ? e : { __proto__: e, toString: (x = e) => this.expr(x) }))
+      return arg
+    })
```

Comment:
Why do we need that in addition to the `{xpr}` handling in `Functions.search()`?

Reply:
`this.tolower(arg)` called for xpr element. becomes `[object Object]` instead of `?` in the sql.